deployment/src/main/java/com/couchbase/quarkus/extension/deployment/CouchbaseDevUiProcessor.java

```diff
@@ -42,26 +43,11 @@ public CardPageBuildItem pages(CouchbaseConfig config) {
      * @return The first hostname, or "localhost" by default.
      */
     private String extractHostnameFromConnectionString(String connectionString) {
-        if (connectionString == null || connectionString.isEmpty()) {
+        ConnectionString connStr = ConnectionString.create(connectionString);
+        if (connStr.hosts().isEmpty()) {
```

Comment:
ConnectionString.hosts() is never empty. ConnectionString.create() throws an exception if you pass a string with zero addresses.